src/coordinator/storage/converter.go

```diff
@@ -177,3 +181,82 @@ func SeriesToPromSamples(series *ts.Series) []*prompb.Sample {
 
 	return samples
 }
+
+const (
+	// TODO(arnikola) get from config
+	initRawFetchAllocSize = 32
+	workerPoolSize        = 10
+)
+
+type decompressRequest struct {
+	iter      encoding.SeriesIterator
+	namespace ident.ID
+	result    *ts.Series
+}
+
+// Process converts the request's SeriesIterator into a ts.Series
+func (w *decompressRequest) Process(ctx context.Context) error {
+	iter := w.iter
+	ns := w.namespace
+	if ns == nil {
+		ns = iter.Namespace()
+	}
+
+	metric, err := FromM3IdentToMetric(ns, iter.ID(), iter.Tags())
+	if err != nil {
+		return err
+	}
+
+	datapoints := make(ts.Datapoints, 0, initRawFetchAllocSize)
+	for iter.Next() {
+		dp, _, _ := iter.Current()
+		datapoints = append(datapoints, ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value})
+	}
+
+	w.result = ts.NewSeries(metric.ID, datapoints, metric.Tags)
+	return nil
+}
+
+// SeriesIteratorsToFetchResult converts SeriesIterators into a fetch result
+func SeriesIteratorsToFetchResult(ctx context.Context, seriesIterators encoding.SeriesIterators, namespace ident.ID) (*FetchResult, error) {
+	defer seriesIterators.Close()
+
+	iters := seriesIterators.Iters()
+	iterLength := seriesIterators.Len()
+
+	seriesList := make([]*ts.Series, 0, iterLength)
+	div, remainder := iterLength/workerPoolSize, iterLength%workerPoolSize
+	var executionPools [][]execution.Request
+	if remainder > 0 {
```

Comment:
i'm hoping we can leverage a library for this instead of doing the math ourselves